experiment/builder_runner.py

```diff
@@ -88,9 +88,12 @@ def _contains_target_function(self, target_path: str) -> bool:
     """Validates if the LLM-generated code contains the target function."""
     with open(target_path) as generated_code_file:
       generated_code = generated_code_file.read()
-    # Get the top level function name without namespace.
-    target_function_name = self.benchmark.function_name.rsplit('::', 1)[-1]
-    return bool(target_function_name in generated_code)
+    # Get the top level function name without namespace or params.
+    pattern = r'(?:[a-zA-Z_]\w*::)*([a-zA-Z_]\w*)(?:\s*<.*>)?\s*\('
+    match = re.search(pattern, self.benchmark.function_signature)
+    min_func_name = (match.group(1).strip()
+                     if match else self.benchmark.function_signature)
+    return bool(min_func_name in generated_code)
```

Comment:
No need for `bool()`. `min_func_name in generated_code` already returns bool.

Reply:
Also, line 110/113 below should use `function_signature` not `function_name`.

Reply:
done